mso/resource_mso_schema_template_contract.go

```diff
@@ -48,511 +50,515 @@ func resourceMSOTemplateContract() *schema.Resource {
+					"voiceAdmit",
+					"unspecified",
+				}, false),
 			},
-			"filter_relationships": {
-				Type: schema.TypeMap,
+			"priority": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"level6",
+					"level5",
+					"level4",
+					"level3",
+					"level2",
+					"level1",
+					"unspecified",
+				}, false),
+			},
+			"filter_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"bothWay",
+					"oneWay",
+				}, false),
+			},
+			"filter_relationship": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-
 						"filter_schema_id": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"filter_template_name": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"filter_name": &schema.Schema{
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"filter_type": &schema.Schema{
 							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"bothWay",
+								"provider_to_consumer",
+								"consumer_to_provider",
+							}, false),
+						},
+						"directives": {
+							Type: schema.TypeSet,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									"none",
+									"no_stats",
+									"log",
+								}, false),
+							},
 							Optional: true,
 							Computed: true,
 						},
-						"filter_template_name": &schema.Schema{
+						"action": {
 							Type:     schema.TypeString,
+							Elem:     &schema.Schema{Type: schema.TypeString},
 							Optional: true,
 							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"deny",
+								"permit",
+							}, false),
 						},
-						"filter_name": &schema.Schema{
+						"priority": {
 							Type:     schema.TypeString,
+							Elem:     &schema.Schema{Type: schema.TypeString},
 							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"level3",
+								"level2",
+								"level1",
+								"default",
+							}, false),
 						},
 					},
 				},
-				Optional:      true,
-				ConflictsWith: []string{"filter_relationship"},
-				Deprecated:    "use filter_relationship instead",
 			},
-			"filter_relationship": {
-				Type:     schema.TypeList,
+			"filter_relationships": {
+				Type:     schema.TypeMap,
```

Comment:
Just for a note, it should be TypeSet or List of Map, as per the current code it stores only the last object information in the state file.

Reply:
correct this is the intention, this is the old format which needed to be set back to original value